Simple Inventory Management System/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,174 @@
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed to delete product: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    public async Task SearchProductAsync(string productName)
+    {
+        const string query = "SELECT Name, Price, Quantity FROM Products WHERE Name = @ProductName";
+        var parameters = new Dictionary<string, object>
+        {
+            { "@ProductName", productName }
+        };
+
+        await DisplayProductListAsync(query, parameters);
+    }
+
+    private async Task<int?> GetProductIdAsync(string productName)
+    {
+        const string query = "SELECT ProductID FROM Products WHERE Name = @ProductName";
+        var parameters = new Dictionary<string, object>
+        {
+            { "@ProductName", productName }
+        };
+
+        var reader = await ExecuteReaderAsync(query, parameters);
+        return await reader.ReadAsync() ? (int?)reader["ProductID"] : null;
+    }
+
+    private async Task ExecuteNonQueryAsync(string query, Dictionary<string, object> parameters, string successMessage)
+    {
+        try
+        {
+            await using var connection = new SqlConnection(connectionString);
+            await using var command = new SqlCommand(query, connection);
+            AddParameters(command, parameters);
+            await connection.OpenAsync();
+            await command.ExecuteNonQueryAsync();
+            DisplaySuccessMessage(successMessage);
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed operation: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    private async Task DisplayProductListAsync(string query, Dictionary<string, object>? parameters = null)
+    {
+        try
+        {
+            await using var reader = await ExecuteReaderAsync(query, parameters);
+            Console.WriteLine("Inventory:");
+            if (!reader.HasRows)
+            {
+                Console.WriteLine("No products found.");
+            }
+            else
+            {
+                while (await reader.ReadAsync())
+                {
+                    Console.WriteLine($"{reader["Name"]} - Price: ${reader["Price"]} - Quantity: {reader["Quantity"]}");
+                }
+            }
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed operation: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+
+    private async Task<SqlDataReader> ExecuteReaderAsync(string query, Dictionary<string, object>? parameters = null)
+    {
+        var connection = new SqlConnection(connectionString);
```

Comment:
assume many reads at the same time, does your code support this case? Search for more better approach